[Flaky Test]: TestSetLogLevelFleetManaged – agent never communicated agent-specific log level "debug" to Fleet

### Failing test case

TestSetLogLevelFleetManaged

### Error message

agent never communicated agent-specific log level "debug" to Fleet

### Build

https://buildkite.com/elastic/elastic-agent-extended-testing/builds/1538#0190cc1c-3e41-42d1-82ba-856491de639e

### OS

Linux

### Stacktrace and notes

```markdown
=== RUN   TestSetLogLevelFleetManaged
    log_level_test.go:372: Creating Agent policy...
    log_level_test.go:376: Creating Agent enrollment API key...
    log_level_test.go:51: Created policy {ID:367ab2ce-b79d-425f-9a7c-38aaad26ded0 Name:testloglevel-policy-d6a4c63a-6a14-4b29-9414-92359c132692 Namespace:default Description:Test Log Level Policy d6a4c63a-6a14-4b29-9414-92359c132692 MonitoringEnabled:[] DataOutputID: MonitoringOutputID: FleetServerHostID: DownloadSourceID: UnenrollTimeout:0 InactivityTImeout:1209600 AgentFeatures:[] IsProtected:false}
    log_level_test.go:53: Getting default Fleet Server URL...
    fixture_install.go:164: [test TestSetLogLevelFleetManaged] Inside fixture install function
    fixture_install.go:192: [test TestSetLogLevelFleetManaged] Inside fixture installNoPkgManager function
    fetcher.go:95: Using existing artifact elastic-agent-8.16.0-SNAPSHOT-linux-x86_64.tar.gz
    fixture.go:282: Extracting artifact elastic-agent-8.16.0-SNAPSHOT-linux-x86_64.tar.gz to /tmp/TestSetLogLevelFleetManaged1237235037/001
    fixture.go:300: Completed extraction of artifact elastic-agent-8.16.0-SNAPSHOT-linux-x86_64.tar.gz to /tmp/TestSetLogLevelFleetManaged1237235037/001
    fixture.go:906: Components were not modified from the fetched artifact
    fixture.go:657: >> running binary with: [/tmp/TestSetLogLevelFleetManaged1237235037/001/elastic-agent-8.16.0-SNAPSHOT-linux-x86_64/elastic-agent install --force --non-interactive --unprivileged --url https://5eac59fc5a6a4375a5a027ffaaab6551.fleet.us-west2.gcp.elastic-cloud.com:443 --enrollment-token aWQ1UXpKQUI0dlg5RXViVjlyelY6eHIwQTd1WUZTRy1lYUhFUUxNMnU4Zw==]
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent status --output json]
    log_level_test.go:210: error fetching agent status: error running command (output: {
            "info": {
                "id": "7db96287-d7d0-4643-81bc-ab6c5fe3c48c",
                "version": "8.16.0",
                "commit": "acfed1974a4dcb265783d5a177b94e97cd3a5720",
                "build_time": "2024-07-19 17:54:23 +0000 UTC",
                "snapshot": true,
                "pid": 30524,
                "unprivileged": true
            },
            "state": 5,
            "message": "Re-executing",
            "components": [],
            "FleetState": 6,
            "FleetMessage": "Not enrolled into Fleet"
        }
        ): exit status 1
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent status --output json]
    log_level_test.go:210: error fetching agent status: could not unmarshal agent status output: error: exit status 1, output: Error: failed to communicate with Elastic Agent daemon: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial unix /opt/Elastic/Agent/elastic-agent.sock: connect: connection refused"
        For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.16/fleet-troubleshooting.html
        
        invalid character 'E' looking for beginning of value
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent status --output json]
    log_level_test.go:210: error fetching agent status: error running command (output: {
            "info": {
                "id": "4b52c472-655d-4f2f-b2dd-585b66e076e2",
                "version": "8.16.0",
                "commit": "acfed1974a4dcb265783d5a177b94e97cd3a5720",
                "build_time": "2024-07-19 17:54:23 +0000 UTC",
                "snapshot": true,
                "pid": 30524,
                "unprivileged": true
            },
            "state": 0,
            "message": "Waiting for initial configuration and composable variables",
            "components": [],
            "FleetState": 0,
            "FleetMessage": ""
        }
        ): exit status 1
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent status --output json]
    log_level_test.go:210: error fetching agent status: error running command (output: {
            "info": {
                "id": "4b52c472-655d-4f2f-b2dd-585b66e076e2",
                "version": "8.16.0",
                "commit": "acfed1974a4dcb265783d5a177b94e97cd3a5720",
                "build_time": "2024-07-19 17:54:23 +0000 UTC",
                "snapshot": true,
                "pid": 30524,
                "unprivileged": true
            },
            "state": 0,
            "message": "Waiting for initial configuration and composable variables",
            "components": [],
            "FleetState": 0,
            "FleetMessage": ""
        }
        ): exit status 1
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent status --output json]
    log_level_test.go:210: error fetching agent status: error running command (output: {
            "info": {
                "id": "4b52c472-655d-4f2f-b2dd-585b66e076e2",
                "version": "8.16.0",
                "commit": "acfed1974a4dcb265783d5a177b94e97cd3a5720",
                "build_time": "2024-07-19 17:54:23 +0000 UTC",
                "snapshot": true,
                "pid": 30524,
                "unprivileged": true
            },
            "state": 0,
            "message": "Waiting for initial configuration and composable variables",
            "components": [],
            "FleetState": 2,
            "FleetMessage": "Connected"
        }
        ): exit status 1
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent status --output json]
    log_level_test.go:219: agent status: {"info":{"id":"4b52c472-655d-4f2f-b2dd-585b66e076e2","version":"8.16.0","commit":"acfed1974a4dcb265783d5a177b94e97cd3a5720","build_time":"2024-07-19 17:54:23 +0000 UTC","snapshot":true,"pid":30524,"unprivileged":true},"state":2,"message":"Running","components":[],"FleetState":2,"FleetMessage":"Connected","upgrade_details":null}
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent inspect]
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent inspect]
    log_level_test.go:89: Setting policy log level to "error"
    log_level_test.go:115: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "info" policy log level: "error"
    log_level_test.go:115: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "info" policy log level: "error"
    log_level_test.go:115: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "info" policy log level: "error"
    log_level_test.go:115: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "info" policy log level: "error"
    log_level_test.go:115: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "info" policy log level: "error"
    log_level_test.go:115: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "info" policy log level: "error"
    log_level_test.go:115: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "info" policy log level: "error"
    log_level_test.go:115: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" policy log level: "error"
    log_level_test.go:123: Setting agent log level to "debug"
    log_level_test.go:248: Updating agent-specific log level to "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:146: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" agent log level: "debug"
    log_level_test.go:140: 
        	Error Trace:	/home/ubuntu/agent/testing/integration/log_level_test.go:140
        	            				/home/ubuntu/agent/testing/integration/log_level_test.go:76
        	Error:      	Condition never satisfied
        	Test:       	TestSetLogLevelFleetManaged
        	Messages:   	agent never communicated agent-specific log level "debug" to Fleet
    log_level_test.go:151: Clearing agent log level, expecting log level to revert back to "error"
    log_level_test.go:248: Updating agent-specific log level to ""
    log_level_test.go:174: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" policy log level: "error"
    log_level_test.go:179: Clearing policy log level, expecting log level to revert back to "info"
    log_level_test.go:202: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" initial log level: "info"
    log_level_test.go:202: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" initial log level: "info"
    log_level_test.go:202: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" initial log level: "info"
    log_level_test.go:202: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "error" initial log level: "info"
    log_level_test.go:202: Fleet metadata log level for agent "4b52c472-655d-4f2f-b2dd-585b66e076e2": "info" initial log level: "info"
    fixture_install.go:265: [test TestSetLogLevelFleetManaged] Inside fixture cleanup function
    fixture_install.go:281: collecting diagnostics; test failed
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent diagnostics -f /home/ubuntu/agent/build/diagnostics/TestSetLogLevelFleetManaged-diagnostics-2024-07-19T18:47:37Z.zip]
    fixture.go:657: >> running binary with: [/opt/Elastic/Agent/elastic-agent uninstall --force]
    fixture.go:1034: Dumping running processes in /home/ubuntu/agent/build/diagnostics/TEST-TestSetLogLevelFleetManaged-linux-amd64-ProcessDump-cleanup.json
--- FAIL: TestSetLogLevelFleetManaged (87.81s)
```

--- a/testing/integration/log_level_test.go
+++ b/testing/integration/log_level_test.go
@@ -41,7 +41,7 @@ func TestSetLogLevelFleetManaged(t *testing.T) {
 		Sudo:  true,
 	})
 
-	deadline := time.Now().Add(10 * time.Minute)
+	deadline := time.Now().Add(30 * time.Minute)
 	ctx, cancel := testcontext.WithDeadline(t, context.Background(), deadline)
 	defer cancel()
 
@@ -115,7 +115,7 @@ func testLogLevelSetViaFleet(ctx context.Context, f *atesting.Fixture, agentID s
 		}
 		t.Logf("Fleet metadata log level for agent %q: %q policy log level: %q", agentID, fleetMetadataLogLevel, policyLogLevel)
 		return fleetMetadataLogLevel == policyLogLevel.String()
-	}, 30*time.Second, time.Second, "agent never communicated policy log level %q to Fleet", policyLogLevel)
+	}, 6*time.Minute, 30*time.Second, "agent never communicated policy log level %q to Fleet", policyLogLevel)
 
 	// Step 2: set a different log level for the specific agent using Settings action
 	// set agent log level and verify that it takes precedence over the policy one
@@ -146,7 +146,7 @@ func testLogLevelSetViaFleet(ctx context.Context, f *atesting.Fixture, agentID s
 		}
 		t.Logf("Fleet metadata log level for agent %q: %q agent log level: %q", agentID, fleetMetadataLogLevel, agentLogLevel)
 		return fleetMetadataLogLevel == agentLogLevel
-	}, 30*time.Second, time.Second, "agent never communicated agent-specific log level %q to Fleet", agentLogLevel)
+	}, 6*time.Minute, 30*time.Second, "agent never communicated agent-specific log level %q to Fleet", agentLogLevel)
 
 	// Step 3: Clear the agent-specific log level override, verify that we revert to policy log level
 	t.Logf("Clearing agent log level, expecting log level to revert back to %q", policyLogLevel)
@@ -174,7 +174,7 @@ func testLogLevelSetViaFleet(ctx context.Context, f *atesting.Fixture, agentID s
 		}
 		t.Logf("Fleet metadata log level for agent %q: %q policy log level: %q", agentID, fleetMetadataLogLevel, policyLogLevel)
 		return fleetMetadataLogLevel == policyLogLevel.String()
-	}, 30*time.Second, time.Second, "agent never communicated reverting to policy log level %q to Fleet", policyLogLevel)
+	}, 6*time.Minute, 30*time.Second, "agent never communicated reverting to policy log level %q to Fleet", policyLogLevel)
 
 	// Step 4: Clear the log level in policy and verify that agent reverts to the initial log level
 	t.Logf("Clearing policy log level, expecting log level to revert back to %q", initialLogLevel)
@@ -202,7 +202,7 @@ func testLogLevelSetViaFleet(ctx context.Context, f *atesting.Fixture, agentID s
 		}
 		t.Logf("Fleet metadata log level for agent %q: %q initial log level: %q", agentID, fleetMetadataLogLevel, initialLogLevel)
 		return fleetMetadataLogLevel == initialLogLevel
-	}, 30*time.Second, time.Second, "agent never communicated initial log level %q to Fleet", initialLogLevel)
+	}, 6*time.Minute, 30*time.Second, "agent never communicated initial log level %q to Fleet", initialLogLevel)
 }
 
 func waitForAgentAndFleetHealthy(ctx context.Context, t *testing.T, f *atesting.Fixture) bool {

